src/ServiceControl.AcceptanceTests/Recoverability/MessageFailures/When_a_failed_message_is_retried.cs

```diff
@@ -0,0 +1,187 @@
+                }).DoNotFailOnErrorMessages())
+                .Done(async ctx =>
+                {
+                    if (ctx.Retried)
+                    {
+                        try
+                        {
+                            await this.Delete($"/api/recoverability/unacknowledgedgroups/{ctx.FailureGroupId}");
+                        }
+                        catch
+                        {
+                            return false;
+                        }
+
+                        failedMessageRetries = await this.TryGet<FailedMessageRetriesCountReponse>("/api/failedmessageretries/count");
+
+                        return failedMessageRetries.Count == 0;
+                    }
+
+                    return false;
+                })
+                .Run();
+
+            Assert.AreEqual(failedMessageRetries.Count, 0, "FaileMessageRetries not removed");
+        }
+
+        [Test]
+        public async Task Should_remove_failedmessageretries_when_retrying_individual_messages()
+        {
+            FailedMessageRetriesCountReponse failedMessageRetries = null;
+
+            await Define<Context>()
+                .WithEndpoint<FailingEndpoint>(b => b.When(async ctx =>
+                {
+                    if (ctx.UniqueMessageId == null)
+                    {
+                        return false;
+                    }
+
+                    FailedMessage failedMessage = await this.TryGet<FailedMessage>($"/api/errors/{ctx.UniqueMessageId}");
+                    if (failedMessage == null)
+                    {
+                        return false;
+                    }
+
+                    return true;
+                }, async (bus, ctx) =>
+                {
+                    ctx.AboutToSendRetry = true;
+                    await this.Post<object>($"/api/errors/{ctx.UniqueMessageId}/retry");
+                }).DoNotFailOnErrorMessages())
+                .Done(async ctx =>
+                {
+                    if (ctx.Retried)
+                    {
+                        failedMessageRetries = await this.TryGet<FailedMessageRetriesCountReponse>("/api/failedmessageretries/count");
+
+                        return failedMessageRetries.Count == 0;
+                    }
+
+                    return false;
+                })
+                .Run();
+
+            Assert.AreEqual(failedMessageRetries.Count, 0, "FaileMessageRetries not removed");
+        }
+
+        public class FailingEndpoint : EndpointConfigurationBuilder
+        {
+            public FailingEndpoint()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.EnableFeature<Outbox>();
+                    c.ReportSuccessfulRetriesToServiceControl();
+
+                    var recoverability = c.Recoverability();
+                    recoverability.Immediate(s => s.NumberOfRetries(1));
```

Comment:
Why do we need a retry?

Reply:
Could this be a left over that got copy-pasted due to the fact that previously tests run across all the transports and we wanted to make sure things don't get deadlettered?

Reply:
Copy-paste indeed